Dockerfile

```diff
@@ -36,12 +36,8 @@ RUN mkdir -p $GOROOT && \
   cd $GOROOT && \
   git checkout release-branch.go1.4
 
-# Patch for skipping a failing test in new docker versions. See https://github.com/getlantern/lantern/issues/2578
-RUN yum install -y patch && yum clean all
-RUN curl https://gist.githubusercontent.com/xiam/f50f6dd6085f9a07ccfd/raw/5e0f472221f9c1556fe34788ff01724b63980337/docker_golang | patch -p0
-
 # Bootstrapping Go.
-RUN cd $GOROOT/src && CGO_ENABLED=1 ./all.bash
+RUN cd $GOROOT/src && CGO_ENABLED=1 ./all.bash || echo "-- Some Go compiler tests failed, continuing."
```

Comment:
We can skip tests at all by using `make.bash` instead, which should speed up the build process a lot.

https://golang.org/doc/install/source#install